src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1021,38 +1021,33 @@ ucs_status_t uct_dc_mlx5_iface_init_fc_ep(uct_dc_mlx5_iface_t *iface)
 
 static void uct_dc_mlx5_iface_cleanup_fc_ep(uct_dc_mlx5_iface_t *iface)
 {
-    uct_dc_mlx5_ep_t *ep = iface->tx.fc_ep;
+    uct_dc_mlx5_ep_t *fc_ep = iface->tx.fc_ep;
     uct_rc_iface_send_op_t *op;
     ucs_queue_iter_t iter;
     uct_rc_txqp_t *txqp;
-    uint8_t num_dcis, dci;
-
-    uct_dc_mlx5_ep_pending_purge(&ep->super.super, NULL, NULL);
-    ucs_arbiter_group_cleanup(&ep->arb_group);
-    uct_rc_fc_cleanup(&ep->fc);
-
-    if (!uct_dc_mlx5_iface_is_dci_rand(iface)) {
-        num_dcis = uct_dc_mlx5_iface_total_ndci(iface);
-
-        for (dci = 0; dci < num_dcis; ++dci) {
-            txqp = &iface->tx.dcis[dci].txqp;
-            ucs_queue_for_each_safe(op, iter, &txqp->outstanding, queue) {
-                if (op->handler == uct_dc_mlx5_ep_fc_pure_grant_send_completion) {
-                    ucs_queue_del_iter(&txqp->outstanding, iter);
-                    op->handler(op, NULL);
-                }
+
+    uct_dc_mlx5_ep_pending_purge(&fc_ep->super.super, NULL, NULL);
+    ucs_arbiter_group_cleanup(&fc_ep->arb_group);
+    uct_rc_fc_cleanup(&fc_ep->fc);
+
+    if (uct_dc_mlx5_iface_is_dci_rand(iface)) {
+        txqp = &iface->tx.dcis[fc_ep->dci].txqp;
+        ucs_queue_for_each_safe(op, iter, &txqp->outstanding, queue) {
+            if (op->handler == uct_dc_mlx5_ep_fc_pure_grant_send_completion) {
+                ucs_queue_del_iter(&txqp->outstanding, iter);
+                op->handler(op, NULL);
             }
         }
-    } else if (ep->dci != UCT_DC_MLX5_EP_NO_DCI) {
-        uct_rc_txqp_purge_outstanding(&iface->super.super,
-                                      &iface->tx.dcis[ep->dci].txqp,
+    } else if (fc_ep->dci != UCT_DC_MLX5_EP_NO_DCI) {
+        txqp = &iface->tx.dcis[fc_ep->dci].txqp;
+        uct_rc_txqp_purge_outstanding(&iface->super.super, txqp,
```

Comment:
comment that in case of dcs, all outstanding operations on this DCI are pure grants

Reply:
done